### ISSUE TITLE
Docker Windows Agent

### ISSUE THREAD

**Issue by cduchenoy** [woodpecker-ci/woodpecker#4274]
### Component

agent

### Describe the bug

From version > 2.6.1 bind mount workspace failed

### Steps to reproduce

1. install woodpecker docker agent (windows) > 2.6.0
2. run an ci

### System Info

```shell
{
source: "https://github.com/woodpecker-ci/woodpecker",
version: "2.6.0"
}
```

### Additional context

First of all: Thank you very much for this useful tool!

For a client we ported the containers (agent, plugin-git) and added one base-step (with chocolatey installed) to Microsoft Windows.

For more information, see <https://github.com/GECO-IT/woodpecker-windows>

We encounter 2 problems:

1 From agent version > 2.6.0, we have a problem mounting volumes in plugins.

Before this release, we could work around the docker error by adding

```yaml
---
...
workspace:
  base: C:\tmp
..
```

We believe this error comes from PR #3933.

2 With the trustedClonePlugins variable <https://github.com/woodpecker-ci/woodpecker/blob/main/shared/constant/constant.go>

We need to retag our local plugin-git image: “woodpeckerci/plugin-git” to keep netrc secure!

Do you think you could include Windows images officially?

For information: we have also include an installer for local backend on Windows.

Best regards
Cyril DUCHENOY
Geco-iT

### Validations

- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]

**Comment by 6543** [comment]
this most likely has to do with https://github.com/woodpecker-ci/woodpecker/pull/3933 witch was done because of https://github.com/woodpecker-ci/woodpecker/issues/3924

**Comment by 6543** [comment]
I would say if workflow workspace has C:\ prefix we do add it there too ...

... feel free to create a pull

**Comment by 6543** [comment]
Hmm also if labels platform has windows ... but in any case this is an heuristic and should be an own function

checkRunOnWindowsHeuristics(...) bool
witch will then add the prefix on true

**Comment by 6543** [comment]
> Do you think you could include Windows images officially?

Do you have an source of your windows git clone plugin ... it's an open issue for our plugin ...

**Comment by cduchenoy** [comment]
Yes in the same repo <https://github.com/GECO-IT/woodpecker-windows/tree/main/plugins/plugin-git>

**Comment by cduchenoy** [comment]
We just dockerize the official plugin under windows

**Comment by cduchenoy** [comment]
for agent https://github.com/GECO-IT/woodpecker-windows/blob/main/agent/backend-docker/Dockerfile.Windows

**Comment by 6543** [comment]
the biggest problem we currently have is: no agent running on windows with docker+windows backend ...

... so we might have a look if we can autoscale windows vms on demand via https://github.com/woodpecker-ci/autoscaler

**Comment by 6543** [comment]
-> https://github.com/woodpecker-ci/autoscaler/issues/133 to only deploy agent with windows if needed based on lable
-> ms azure support on the autoscaler

another option would be if somebody sponsor an agent that run all the time ... but i would call it an hotfix

**Comment by cduchenoy** [comment]
Hi,

I updated our repository <https://github.com/GECO-IT/woodpecker-windows>

As expected, I submitted our [teams-notify](https://github.com/woodpecker-ci/woodpecker/pull/4400) plugin :)

I would also like to prepare the expected PRs but I don't really know what to send...

I just uploaded the docker file for the agent and plugin?

- Agent: https://github.com/GECO-IT/woodpecker-windows/blob/main/agent/backend-docker/Dockerfile.Windows
- Git plugin: <https://github.com/GECO-IT/woodpecker-windows/blob/main/plugins/plugin-git/Dockerfile.Windows>

**Comment by johanvdw** [comment]
Thanks for your work on the windows support!

I was able to make most things work. Some comments that might be useful for other people wanting to try this:
* as soon as you have a windows agent configured, it's a good idea to add 
```
labels:
  platform: linux/amd64
  backend: docker
``` 
to any pipelines you want to run under linux and not windows.  Because windows agent may pick it up and then fail because "the image is not found" which is a strange error (the image exists, but not for windows) and it may not be immediately clear that another agent picked this up.
* I had issues with git-lfs on windows. Looks like the image uses .netrc and not _netrc, which git-lfs requires. I was able to fix using a custom git-lfs build ( https://github.com/git-lfs/git-lfs/pull/6055 )
* Anyway, we have used woodpecker on windows, and it works nicely. I'm happy to help testing/integrating more of this.

**Comment by cduchenoy** [comment]
Great that it worked for you!

The image wasn't found because there's no official Woodpecker image for the git plugin on Windows...

I opened an issue: https://github.com/woodpecker-ci/plugin-git/issues/187
The current Woodpecker team doesn't have the resources to maintain and test it on Windows.

Thanks for your feedback.
/Cyril
